app/models/user.rb

```diff
@@ -0,0 +1,28 @@
+class User < ApplicationRecord
+  validates :name,  presence: true, length: { maximum: 30 }
+  validates :email, presence: true, length: { maximum: 255 },
+                    format: { with: /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i },
+                    uniqueness: true
+  before_validation { email.downcase! }
+  has_secure_password
+  validates :password, presence: true, length: { minimum: 6 }, allow_nil: true
+
+  has_many :tasks, dependent: :destroy
+
+  before_destroy :ensure_admin_deatroy
+  before_update :ensure_admin_update
+
+  private
+  def ensure_admin_deatroy
```

Comment:
タイプミスしてしまっているかと思います。
```suggestion
  def ensure_admin_destroy
```